xscen/extract.py

```diff
@@ -1078,3 +1084,151 @@ def _subset_file_coverage(
         files_to_keep = files_to_keep | files_in_range
 
     return df[files_to_keep]
+
+
+@parse_config
+def prepare_warming_level(
+    ds: xr.Dataset,
+    wl: float,
+    window: int = 20,
+    tas_baseline_period: list = None,
+    ignore_member: bool = False,
+    tas_csv: str = None,
+    to_level: str = "warminglevel-{wl}",
+    wl_dim: str = "+{wl}Cvs{period0}-{period1}",
+):
+    """
+    Returns the input dataset with only the window of time over which the given level of
+    global warming is first reached using the IPCC Atlas method.
+
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+      Input dataset.
+      The dataset should include attributes to help recognize it and find its
+      warming levels - 'cat:mip_era', 'cat:experiment', 'cat:member', and either
+      'cat:source' for global models or 'cat:driving_model' for regional models.
+    wl: float
+      Warming level.
+      eg. 2 for a global warming level of +2 degree Celsius above the mean temperature of the `tas_baseline_period`.
+    window: int
+      Size of the rolling window in years over which to compute the warming level.
+    tas_baseline_period: list
+      Base period. The warming is calculated with respect to this period. The default is ["1850", "1900"].
+    ignore_member: bool
+      Whether to ignore the member when searching for the model run in tas_csv.
+    tas_csv: str
+      Path to a csv of annual global mean temperature with a row for each year and a column for each dataset.
+      If None, it will default to data/IPCC_annual_global_tas.csv which was built from
+      the IPCC atlas data from  Iturbide et al., 2020 (https://doi.org/10.5194/essd-12-2959-2020)
+      and extra data from pilot models of MRCC5 and ClimEx.
+    to_level:
+      The processing level to assign to the output.
+      Use "{wl}", "{period0}" and "{period1}" in the string to dynamically include
+      `wl`, 'tas_baseline_period[0]' and 'tas_baseline_period[1]'.
+    wl_coord: str
```

Comment:
Discussion: Do we want it as a `coord` or `dim`? I'd tend to go for a `coord`, but I don't have a strong opinion. One point going for `dim` is that it allows using `sel / squeeze / etc.`

(Also, the name in the Docstrings doesn't match the variable)

Reply:
I'll fix the name. I would leave it as a dim, because if you have many wl, you can concat them together and it has to be a dim in that case.